sonar-scanner-engine/src/main/java/org/sonar/scanner/sensor/DefaultSensorStorage.java

```diff
@@ -113,71 +118,148 @@ public DefaultSensorStorage(MetricFinder metricFinder, ModuleIssues moduleIssues
     this.measureCache = measureCache;
     this.index = index;
     this.contextPropertiesCache = contextPropertiesCache;
-  }
+    this.scannerMetrics = scannerMetrics;
 
-  private Metric findMetricOrFail(String metricKey) {
-    Metric m = (Metric) metricFinder.findByKey(metricKey);
-    if (m == null) {
-      throw new IllegalStateException("Unknow metric with key: " + metricKey);
-    }
-    return m;
+    coverageMetrics.add(CoreMetrics.UNCOVERED_LINES);
+    coverageMetrics.add(CoreMetrics.LINES_TO_COVER);
+    coverageMetrics.add(CoreMetrics.UNCOVERED_CONDITIONS);
+    coverageMetrics.add(CoreMetrics.CONDITIONS_TO_COVER);
+    coverageMetrics.add(CoreMetrics.CONDITIONS_BY_LINE);
+    coverageMetrics.add(CoreMetrics.COVERED_CONDITIONS_BY_LINE);
+    coverageMetrics.add(CoreMetrics.COVERAGE_LINE_HITS_DATA);
+
+    byLineMetrics.add(CoreMetrics.COVERAGE_LINE_HITS_DATA);
+    byLineMetrics.add(CoreMetrics.COVERED_CONDITIONS_BY_LINE);
+    byLineMetrics.add(CoreMetrics.CONDITIONS_BY_LINE);
+
+    deprecatedCoverageMetricMapping.put(CoreMetrics.IT_COVERAGE, CoreMetrics.COVERAGE);
+    deprecatedCoverageMetricMapping.put(CoreMetrics.IT_LINE_COVERAGE, CoreMetrics.LINE_COVERAGE);
+    deprecatedCoverageMetricMapping.put(CoreMetrics.IT_BRANCH_COVERAGE, CoreMetrics.BRANCH_COVERAGE);
+    deprecatedCoverageMetricMapping.put(CoreMetrics.IT_UNCOVERED_LINES, CoreMetrics.UNCOVERED_LINES);
+    deprecatedCoverageMetricMapping.put(CoreMetrics.IT_LINES_TO_COVER, CoreMetrics.LINES_TO_COVER);
+    deprecatedCoverageMetricMapping.put(CoreMetrics.IT_UNCOVERED_CONDITIONS, CoreMetrics.UNCOVERED_CONDITIONS);
+    deprecatedCoverageMetricMapping.put(CoreMetrics.IT_CONDITIONS_TO_COVER, CoreMetrics.CONDITIONS_TO_COVER);
+    deprecatedCoverageMetricMapping.put(CoreMetrics.IT_CONDITIONS_BY_LINE, CoreMetrics.CONDITIONS_BY_LINE);
+    deprecatedCoverageMetricMapping.put(CoreMetrics.IT_COVERED_CONDITIONS_BY_LINE, CoreMetrics.COVERED_CONDITIONS_BY_LINE);
+    deprecatedCoverageMetricMapping.put(CoreMetrics.IT_COVERAGE_LINE_HITS_DATA, CoreMetrics.COVERAGE_LINE_HITS_DATA);
+    deprecatedCoverageMetricMapping.put(CoreMetrics.OVERALL_COVERAGE, CoreMetrics.COVERAGE);
+    deprecatedCoverageMetricMapping.put(CoreMetrics.OVERALL_LINE_COVERAGE, CoreMetrics.LINE_COVERAGE);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "OVERALL_LINE_COVERAGE"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)